README.md

```diff
@@ -503,25 +592,25 @@ The following UCAN fragment would be valid to invoke as `did:key:zH3C2AVvLMv6gmM
 
-Except for rights amplification (below), each capability delegation MUST have equal or narrower capabilities from its witnesses. The time bounds MUST also be equal to or contained inside the time bounds of the witnesses' time bounds. This lowering of rights at each delegation is called "attenuation."
+Except for rights amplification (below), each capability delegation MUST have equal or narrower capabilities from its proofs. The time bounds MUST also be equal to or contained inside the time bounds of the proof's time bounds. This lowering of rights at each delegation is called "attenuation."
 
 ## 5.4 Rights Amplification
 
-Some capabilities are more than the sum of their parts. The canonical example is a can of soup and a can opener. You need both to access the soup inside the can, but the can opener may come from a completely separate source than the can of soup. Such semantics MAY be implemented in UCAN capabilities. This means that validating particular capabilities MAY require more than one direct witness. The relevant witnesses MAY be of a different resource and action from the amplified capability. The delegated capability MUST have this behaviour in its semantics, even if the witnesses do not.
+Some capabilities are more than the sum of their parts. The canonical example is a can of soup and a can opener. You need both to access the soup inside the can, but the can opener may come from a completely separate source than the can of soup. Such semantics MAY be implemented in UCAN capabilities. This means that validating particular capabilities MAY require more than one direct proof. The relevant proofs MAY be of a different resource and action from the amplified capability. The delegated capability MUST have this behaviour in its semantics, even if the proofs do not.
 
 ## 5.6 Content Identifiers
 
-UCANs MAY be referenced by content ID (CID), per the [multiformats/cid](https://github.com/multiformats/cid) specification. The resolution of these addresses is left to the implementation and end-user, and MAY (non-exclusively) includes the following: local store, a distributed hash table (DHT), gossip network, or RESTful service.
+A UCAN token MUST be referenced as a [base32](https://github.com/multiformats/multibase/blob/master/multibase.csv#L12) [CIDv1](https://docs.ipfs.io/concepts/content-addressing/#identifier-formats), using either the [`0x55` raw data](https://github.com/multiformats/multicodec/blob/master/table.csv#L39) or [`0x0129` `dag-json`](https://github.com/multiformats/multicodec/blob/master/table.csv#L104) multicodec. Note that if using `dag-json`, the UCAN MUST be formatted as `dag-json` prior to signing.
```

Comment:
@Gozala the previous comment I made to https://github.com/ucan-wg/spec/pull/55#discussion_r886956218 disappeared when I changed this line.

TL;DR this used to have dag-ucan, but I removed that line when the PR didn't go forward. After we decided to use dag-json, I forgot to reference that.

How does this revised with `dag-json` look?

Reply:
> TL;DR this used to have dag-ucan, but I removed that line when the PR didn't go forward. After we decided to use dag-json, I forgot to reference that.
>
> How does this revised text with dag-json look?

I am personally interested in adding dag-cbor here which is what `dag-ucan` uses. I suppose `dag-json` could also be included, that said I am not sure if it would provide any value given that `dag-cbor` is more compact and faster to work with.

I'm also not sure how useful this note is

> Note that if using `dag-json`, the UCAN MUST be formatted as `dag-json` prior to signing.

It probably should specify IPLD schema instead

Reply:
My memory is jogging on this, which may lead to a total (but short!) rewrite of this section. We determined that we can ask the `dag-cbor` version to sign with a regular JWT method, which is where I think my wires got crossed here (we've been talking about e.g. EIP-191 signature support in the SIWE/CACAO discussion).

`dag-cbor`, `dag-json`, `dag-pb`, etc enforce deterministic encoding, which means that they can also always be turned into a plain ol' JWT. If they're signed on the JWT version, we can store them however we want.

@Gozala just triple checking that the above is correct. 👍 / 👎 ?

If so, the spec should specify that a nondeterministic encoder MUST implementation MUST produce `0x55` "raw" CIDs, but leave arbitrary other codecs open. As long as a client can request them in a JWT encoding (which would be left up to the transport spec), Web 2.0-focused clients don't need to ship IPLD in their app, and it leaves it open to arbitrary IPFD codecs (which even IPLD-enabled clients may not have access to all of them). This way, the system overall follows the bit of Postel's Law where you're "liberal in what you accept". A client not equipped with the relevant IPLD codec wouldn't be able to validate the hash, but:

1. IPLD clients already often skip that step
3. UCAN is proven constructively, so any satisfying proof should work (aligned iss/aud, caps correct, etc).

How does that sound?

Any thoughts on standardizing on CIDv1 and base32 for the IPLD clients + always being HTTP safe?

---

<sub>Aside: #2 above suggests that if we later change the format away from JWT, the proofs don't actually need to be signed 🤔 There may be something obvious about why that's a terrible idea, but nothing is coming to mind right now</sub>

Reply:
> dag-cbor, dag-json, dag-pb, etc enforce deterministic encoding, which means that they can also always be turned into a plain ol' JWT. If they're signed on the JWT version, we can store them however we want.
>
> @Gozala just triple checking that the above is correct. 👍 / 👎 ?

Precisely 

> If so, the spec should specify that a nondeterministic encoder MUST implementation MUST produce 0x55 "raw" CIDs, but leave arbitrary other codecs open. As long as a client can request them in a JWT encoding (which would be left up to the transport spec), Web 2.0-focused clients don't need to ship IPLD in their app, and it leaves it open to arbitrary IPFD codecs (which even IPLD-enabled clients may not have access to all of them). This way, the system overall follows the bit of Postel's Law where you're "liberal in what you accept". 

I'm ok with not prescribing specific encoding and leaving it open ended. I still think there is a benefit on rallying around specific IPLD schema though, but that can come in a next revision of the protocol once things settle there.

>  A client not equipped with the relevant IPLD codec wouldn't be able to validate the hash, but:
>
>    - IPLD clients already often skip that step
>    - UCAN is proven constructively, so any satisfying proof should work (aligned iss/aud, caps correct, etc).
>
> How does that sound?

I'm not sure I understand the concern here. Do you imply that verifier will get JWT encoding of the proof while it is linked with e.g. with CBOR CID ?

If so it is correct that verifier would not be able to verify that CID actually corresponds to the provided proof without having a codec. But then again it seems like a really bad idea to link in one encoding but transmit in the other to an actor that is unable to translate between two.

> Any thoughts on standardizing on CIDv1 and base32 for the IPLD clients + always being HTTP safe?

Anything other that UnixFS requires CIDv1 so that is MUST. Initially I was going to say it's better to keep the encoding open, but then I realized that this would also affect signatures and therefor it would be best to prescribe it as well.

> always being HTTP safe

I'm not sure I understand what you mean here.

> Aside: https://github.com/ucan-wg/spec/issues/2 above suggests that if we later change the format away from JWT, the proofs don't actually need to be signed 🤔 There may be something obvious about why that's a terrible idea, but nothing is coming to mind right now

I'm not I understand how did you arrive to "proofs don't actually need to be signed" ? If that is a case that indeed seems like a really bad idea.